queue-manager/core/src/manager.ts

```diff
@@ -277,6 +277,28 @@ class Manager {
     this.queues.delete(queue_id);
     void this.persistor.deleteQueue(queue_id);
   }
+
+  public async clearQueue() {
+    const queuesToDelete: QueueID[] = [];
+
+    // Identify queues that are not running
+    this.queues.forEach((queue, queue_id) => {
```

Comment:
I think you can use `filter` here to be more performant considering it benefits from internal optimizations. Or maybe you can delete in one iteration instead of pushing each item in a new array and delete them in another iteration.